src/main/scala/org/codeoverflow/chatoverflow/requirement/parameter/ColorParameterImpl.scala

```diff
@@ -0,0 +1,53 @@
+package org.codeoverflow.chatoverflow.requirement.parameter
+
+import org.codeoverflow.chatoverflow.api.io.parameter.ColorParameter
+import org.codeoverflow.chatoverflow.registry.Impl
+import java.awt.Color
+
+/**
+  * A parameter holding a Color value.
+  */
+@Impl(impl = classOf[ColorParameter])
+class ColorParameterImpl extends ColorParameter {
+  private var value: Color = null
+
+  override def getType: Class[Color] = classOf[Color]
+
+  override def serialize(): String = s"${value.getRed},${value.getGreen},${value.getBlue},${value.getAlpha}"
+
+  override def get(): Color = value
+
+  override def deserialize(value: String): Unit = {
+
+    val hex3 = "^#([a-fA-F0-9]{6})$".r
+    val hex4 = "^#([a-fA-F0-9]{8})$".r
+    val int3 = "^(\\d+),(\\d+),(\\d+)$".r
+    val int4 = "^(\\d+),(\\d+),(\\d+),(\\d+)$".r
+    val float3 = "^(\\d+(?:\\.\\d+)?),(\\d+(?:\\.\\d+)?),(\\d+(?:\\.\\d+)?)$".r
+    val float4 = "^(\\d+(?:\\.\\d+)?),(\\d+(?:\\.\\d+)?),(\\d+(?:\\.\\d+)?),(\\d+(?:\\.\\d+)?)$".r
```

Comment:
It also would be nice to have some comments here that help to understand which regex is for which syntax.

Reply:
the name of the regex already says that:
int3 -> 3 ints
float4 -> 4 floats